configure cross-compiling fails in new versions

### Description
Using this command:
```../configure --disable-shared --disable-debug --host=arm-linux-gnueabihf```
**_configure_** fails in version 3.2.1 and 3.1.7, while it succeeds with version 3.1.4 and older.


**_configure_** fails in testing GTK:

>   checking for GTK+ version...
>   checking for arm-linux-gnueabihf-pkg-config... (cached) /usr/bin/arm-linux-gnueabihf-pkg-config
>   checking pkg-config is at least version 0.16... yes
>   checking for GTK+ - version >= 3.0.0... Package gtk+-3.0 was not found in the pkg-config search path.
>   Perhaps you should add the directory containing `gtk+-3.0.pc'
>   to the PKG_CONFIG_PATH environment variable
>   No package 'gtk+-3.0' found
>   Package gthread-2.0 was not found in the pkg-config search path.
>   Perhaps you should add the directory containing `gthread-2.0.pc'
>   to the PKG_CONFIG_PATH environment variable

and after other errors:

> *** Could not run GTK+ test program, checking why...
*** The test program failed to compile or link. See the file config.log for the
*** exact error that occurred. This usually means GTK+ is incorrectly installed.
checking for gtk-config... no
checking for GTK - version >= 1.2.7... no
*** The gtk-config script installed by GTK could not be found
*** If GTK was installed in PREFIX, make sure PREFIX/bin is in
*** your path, or set the GTK_CONFIG environment variable to the
*** full path to gtk-config.
checking for gtk-config... (cached) no
checking for GTK - version >= 1.2.3... no
*** The gtk-config script installed by GTK could not be found
*** If GTK was installed in PREFIX, make sure PREFIX/bin is in
*** your path, or set the GTK_CONFIG environment variable to the
*** full path to gtk-config.
configure: error:
The development files for GTK+ were not found. For GTK+ 2, please
ensure that pkg-config is in the path and that gtk+-2.0.pc is
installed. For GTK+ 1.2 please check that gtk-config is in the path,
and that the version is 1.2.3 or above. Also check that the
libraries returned by 'pkg-config gtk+-2.0 --libs' or 'gtk-config
--libs' are in the LD_LIBRARY_PATH or equivalent.

The same command works flawlessy with wx version 3.1.4 and 3.1.2; i then can build the widgets and the _minimal_ sample app.


### Platform and version information

- wxWidgets version 3.2.1 and 3.1.7, source files for linux.
- tested and verified on two PCs: one Debian11 in a Windows 10 WSL 2, one Debian 9 32bit + Gnome
- Compiler being used: gcc

> So it looks like using `/usr/bin/arm-linux-gnueabihf-pkg-config` should work, thanks. Where does this file come from BTW? I don't see it in (non-ARM) Debian repositories. I'd like to install it here to be able to test myself.

these should be the relevant commands from _history_:
```
sudo apt-get install libgtk-3-dev build-essential checkinstall
sudo apt install gcc make gcc-arm-linux-gnueabihf binutils-arm-linux-gnueabihf
sudo dpkg --add-architecture armhf
sudo apt-get update
sudo apt install libgtk-3-dev:armhf
```


> Not quite,

Sorry. Let me try again:
**v3.1.4**
```
configure:21330: checking for arm-linux-gnueabihf-pkg-config
configure:21348: found /usr/bin/arm-linux-gnueabihf-pkg-config
configure:21360: result: /usr/bin/arm-linux-gnueabihf-pkg-config
configure:21428: checking pkg-config is at least version 0.9.0
configure:21431: result: yes

...

configure:22875: checking for GTK+ version
configure:22884: result: 
configure:22935: checking for arm-linux-gnueabihf-pkg-config
configure:22965: result: /usr/bin/arm-linux-gnueabihf-pkg-config
configure:23033: checking pkg-config is at least version 0.16
configure:23036: result: yes
configure:23050: checking for GTK+ - version >= 3.0.0
configure:23171: result: yes (version 3.24.24)
configure:26298: checking for X11/Xlib.h
configure:26298: arm-linux-gnueabihf-gcc -c -D_GNU_SOURCE   conftest.c >&5
configure:26298: $? = 0
configure:26298: result: yes
configure:26311: checking for X11/XKBlib.h
configure:26311: arm-linux-gnueabihf-gcc -c -D_GNU_SOURCE   conftest.c >&5
configure:26311: $? = 0
configure:26311: result: yes
configure:26715: checking for arm-linux-gnueabihf-pkg-config
configure:26745: result: /usr/bin/arm-linux-gnueabihf-pkg-config
configure:26813: checking pkg-config is at least version 0.9.0
configure:26816: result: yes

```

**v3.1.7**
```
configure:25712: checking for arm-linux-gnueabihf-pkg-config
configure:25730: found /usr/bin/arm-linux-gnueabihf-pkg-config
configure:25742: result: /usr/bin/arm-linux-gnueabihf-pkg-config
configure:25810: checking pkg-config is at least version 0.9.0
configure:25813: result: yes

...

configure:27572: checking for GTK+ version
configure:27581: result: 
configure:27633: checking for arm-linux-gnueabihf-pkg-config
configure:27663: result: /usr/bin/arm-linux-gnueabihf-pkg-config
configure:27731: checking pkg-config is at least version 0.16
configure:27734: result: yes
configure:27748: checking for GTK+ - version >= 3.0.0
configure:27875: result: no
configure:27903: arm-linux-gnueabihf-gcc -o conftest -D_GNU_SOURCE     conftest.c  -lm  >&5
conftest.c:53:10: fatal error: gtk/gtk.h: No such file or directory
   53 | #include <gtk/gtk.h>
      |          ^~~~~~~~~~~
compilation terminated.

```
Hope this is right :-|

> > There's a temp solution, or even a hack, i could use to be able to bypass the problem?
> You could try reverting the commit mentioned above -- does it work for you?

I really don't know how to do it :|
Does it means that i could overwrite the newer _configure.*_ with the older, working ones?

--- a/.github/workflows/ci_mac.yml
+++ b/.github/workflows/ci_mac.yml
@@ -93,6 +93,7 @@ jobs:
           runner: macos-11
           arch: x86_64
           configure_flags: --with-osx_iphone --enable-monolithic  --disable-sys-libs --host=i686-apple-darwin_sim --build=x86_64-apple-darwin17.7.0
+          configure_env: PKG_CONFIG_LIBDIR=/dev/null
           xcode_sdk: iphonesimulator
           skip_samples: true
           skip_testing: true
@@ -171,9 +172,9 @@ jobs:
           wxASAN_CXXFLAGS=$wxASAN_CFLAGS
           wxASAN_LDFLAGS="-fsanitize=address"
 
-          ./configure $wxCONFIGURE_OPTIONS --enable-debug "CFLAGS=$wxASAN_CFLAGS" "CXXFLAGS=$wxASAN_CXXFLAGS" "LDFLAGS=$wxASAN_LDFLAGS" || rc=$?
+          ${{ matrix.configure_env }} ./configure $wxCONFIGURE_OPTIONS --enable-debug "CFLAGS=$wxASAN_CFLAGS" "CXXFLAGS=$wxASAN_CXXFLAGS" "LDFLAGS=$wxASAN_LDFLAGS" || rc=$?
         else
-          ./configure $wxCONFIGURE_OPTIONS --disable-debug_info || rc=$?
+          ${{ matrix.configure_env }} ./configure $wxCONFIGURE_OPTIONS --disable-debug_info || rc=$?
         fi
         if [ -n "$rc" ]; then
           echo '*** Configuring failed, contents of config.log follows: ***'

--- a/.github/workflows/ci_mac.yml
+++ b/.github/workflows/ci_mac.yml
@@ -99,7 +99,6 @@ jobs:
           allow_warnings: true
 
     env:
-      wxCONFIGURE_FLAGS: --disable-sys-libs --without-liblzma ${{ matrix.configure_flags }} --prefix=${{ github.workspace }}/localbin_${{ matrix.arch }}
       wxUSE_ASAN: ${{ matrix.use_asan && 1 || 0 }}
       DEVELOPER_DIR: /Applications/Xcode.app/Contents/Developer
       NSUnbufferedIO: YES
@@ -160,7 +159,7 @@ jobs:
 
     - name: Configuring
       run: |
-        wxCONFIGURE_OPTIONS="--disable-optimise $wxCONFIGURE_FLAGS"
+        wxCONFIGURE_OPTIONS="--disable-optimise --disable-sys-libs --without-liblzma ${{ matrix.configure_flags }} --prefix=${{ github.workspace }}/localbin_${{ matrix.arch }}"
 
         if [ -n "${{ matrix.xcode_sdk }}" ]; then
           sdk_path=`xcrun --sdk ${{ matrix.xcode_sdk }} --show-sdk-path`

--- a/configure
+++ b/configure
@@ -25188,15 +25188,6 @@ $as_echo "no" >&6; }
 
 fi
 
-if test "$build" != "$host"; then
-
-
-                if test -z "$PKG_CONFIG_LIBDIR"; then
-        PKG_CONFIG_LIBDIR=/dev/null
-        export PKG_CONFIG_LIBDIR
-    fi
-fi
-
 
 
 if test "$wxUSE_REGEX" != "no"; then

--- a/configure.in
+++ b/configure.in
@@ -2106,23 +2106,6 @@ fi
 dnl Find pkg-config outside of any conditional.  Done before any PKG_* call.
 PKG_PROG_PKG_CONFIG
 
-dnl When cross-compiling, don't use .pc files on the build system, they are at
-dnl best useless and can be harmful (e.g. they may define options inappropriate
-dnl for the cross-build, resulting in the failure of all the subsequent tests).
-if test "$build" != "$host"; then
-    dnl pkg.m4 forbids the use of PKG_XXX, so undo it here to avoid autoconf
-    dnl errors.
-    m4_pattern_allow([PKG_CONFIG_LIBDIR])
-
-    dnl If pkg-config libdir is already defined, we suppose that they know what
-    dnl they're doing and leave it alone, but if not, set it to a path in which
-    dnl no .pc files will be found.
-    if test -z "$PKG_CONFIG_LIBDIR"; then
-        PKG_CONFIG_LIBDIR=/dev/null
-        export PKG_CONFIG_LIBDIR
-    fi
-fi
-
 dnl ---------------------------------------------------------------------------
 dnl Optional libraries
 dnl

